tcms/core/ajax.py

```diff
@@ -137,23 +136,23 @@ def tags(self):
         def users(self):
             from django.contrib.auth.models import User
 
-            query = strip_parameters(self.request, self.internal_parameters)
+            query = strip_parameters(self.request.GET, self.internal_parameters)
             return User.objects.filter(**query)
 
         def versions(self):
             from tcms.management.models import Version
 
             return Version.objects.filter(product__id=self.product_id)
 
-    objects = Objects(request=request, product_id=request.GET['product_id'])
+    objects = Objects(request=request, product_id=request.GET.get('product_id'))
```

Comment:
As mentioned https://github.com/Nitrate/Nitrate/pull/178/commits/8ca779b308b31922bc00fed3b598843e6a78b67f#r116201023, please move other unrelated changes to separate patches.